include/zephyr/net/http/server.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+#define ZEPHYR_INCLUDE_NET_HTTP_SERVER_H_
+
+#if !defined(__ZEPHYR__) || defined(CONFIG_POSIX_API)
+
+#include <stdint.h>
+#include <poll.h>
+
+#else
+
+#include <zephyr/posix/poll.h>
+
+#endif
+
+#if !defined(__ZEPHYR__)
+
+#define CONFIG_NET_HTTP2_MAX_CLIENTS 10
+#define CONFIG_NET_HTTP2_MAX_STREAMS 5
+#define CONFIG_NET_HTTP2_CLIENT_BUFFER_SIZE 256
+
+#endif
+
+#define CLIENT_BUFFER_SIZE CONFIG_NET_HTTP2_CLIENT_BUFFER_SIZE
+#define MAX_CLIENTS CONFIG_NET_HTTP2_MAX_CLIENTS
+#define MAX_STREAMS CONFIG_NET_HTTP2_MAX_STREAMS
+
+struct http2_server_config {
+	int port;
+	int address_family;
+};
+
+enum http2_server_state {
+	HTTP2_FRAME_HEADER_STATE,
+	HTTP_PREFACE_STATE,
+	HTTP1_REQUEST_STATE,
+	HTTP2_FRAME_DATA_STATE,
+	HTTP2_FRAME_HEADERS_STATE,
+	HTTP2_FRAME_SETTINGS_STATE,
+	HTTP2_FRAME_PRIORITY_STATE,
+	HTTP2_FRAME_WINDOW_UPDATE_STATE,
+	HTTP2_FRAME_CONTINUATION_STATE,
+	HTTP2_FRAME_PING_STATE,
+	HTTP2_FRAME_RST_STREAM_STATE,
+	HTTP2_FRAME_GOAWAY_STATE,
+	HTTP_DONE_STATE,
+};
+
+enum http2_streaming_state {
+	IDLE_STATE,
+	OPEN_STATE,
+	CLOSE_STATE
```

Comment:
I would highly recommend that http-server-specific enumerations be namespaced - e.g. `HTTP_SERVER_FOO`.